apps/myaccount/src/layouts/dashboard.tsx

```diff
@@ -167,15 +168,24 @@ export const DashboardLayout: FunctionComponent<PropsWithChildren<DashboardLayou
         }
 
         const routes: RouteInterface[] = getDashboardLayoutRoutes().filter((route: RouteInterface) => {
-            if (route.path === AppConstants.getPaths().get("APPLICATIONS") && !isApplicationsPageVisible) {
-                return false;
+
+            if (allowedScopes.includes("internal_user_impersonate")) {
```

Comment:
Let's add a comment here explaining why this filtering is necessary. also add that there should be a way to handle dynamic route filtering.